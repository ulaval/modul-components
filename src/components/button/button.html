<button :class="[ componentName,
                { 'is-primary': type!='secondary',
                'is-secondary': type=='secondary',
                'is-selected': state=='selected',
                'is-waiting': state=='waiting',
                'is-disabled': state=='disabled' } ]"
        v-m-ripple-effect="!isDisabled"
        :disabled="isDisabled">
    <span :class="[ componentName + '__icon', 'is-left']" v-if="hasIconeLeft">
        <slot name="icon-left"></slot>
    </span>
    <span :class="[ componentName + '__text']">
        <slot>{{ componentName }}</slot>
        <span :class="[ componentName + '__more-info' ]" v-if="hasMoreInfo">
            <slot name="more-info"></slot>
        </span>
    </span>
    <span :class="[ componentName + '__icon', 'is-right']" v-if="hasIconeRight">
        <slot name="icon-right"></slot>
    </span>
</button>