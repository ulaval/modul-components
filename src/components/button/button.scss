
@import '../../styles/abstracts/variables';

$m-button--color: $m-color--primary !default;
$m-button--color--hover: $m-color--primary--light !default;
$m-button--color--active: $m-color--primary--darker !default;
$m-button--color--waiting: $m-color--primary--lighter !default;
$m-button--color--disabled: $m-color--grey--light !default;

.m-button {
    position: relative;

    display: inline-flex;
    align-items: center;

    text-align: center;
    vertical-align: middle;
    padding: 10px $m-padding 9px $m-padding;



    appearance: none;
    outline: none;
    user-select: none;
    text-decoration: none;

    font-size: $m-font-size;
    font-weight: $m-font-weight--bold;
    line-height: 1;

    border: 1px solid $m-button--color;
    border-radius: 4px;

    &.is-primary,
    &.is-secondary {
        cursor: pointer;

        &:not(.is-waiting):not(.is-disabled) {
            &:hover {
                color: $m-color--white;
                background: $m-button--color--hover;
                border-color: $m-button--color--hover;
            }
            &:active {
                background: $m-button--color--active;
                border-color: $m-button--color--active;
            }
        }
    }

    &.is-primary {
        color: $m-color--white;
        background: $m-button--color;
    }

    &.is-secondary {
        background: none;
        color: $m-button--color;
    }

    &.is-waiting,
    &.is-disabled {
        transition: none;
    }

    &.is-waiting {
        cursor: progress;
        color: $m-color--white;
        background: $m-button--color--waiting;
        border-color: $m-button--color--waiting;
    }

    &.is-disabled {
        cursor: not-allowed;
        color: $m-button--color--disabled;
        background: $m-color--white;
        border-color: $m-button--color--disabled;
    }
}

.m-button__text {
    flex: 1;
}

.m-button__more-info,
.m-button__icon {
    &:empty {
        display: none;
    }
}

.m-button__more-info {
    display: block;

    font-size: $m-font-size--s;
    font-weight: $m-font-weight--regular;

    margin-top: $m-margin--s;

    &:empty {
        display: none;
    }
}

.m-button__icon {
    &.is-left {
        margin-right: $m-padding--s;
    }
    &.is-right{
        margin-left: $m-padding--s;
    }
}


