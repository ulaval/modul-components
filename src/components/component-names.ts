export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
