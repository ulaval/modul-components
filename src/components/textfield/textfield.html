<div class="m-textfield"
     :class="[{ 'm--is-disabled' : isDisabled,
                'm--is-waiting' : isWaiting,
                'm--has-error' : hasError,
                'm--is-valid' : isValid,
                'm--has-icon' : hasIcon,
                'm--has-label': hasLabel },
                'm--is-type-' + inputType ]"
     @click="onClick"
     :style="{ width: inputWidth, maxWidth: inputMaxWidth }">
    <m-input-style :disabled="isDisabled"
                   :waiting="isWaiting"
                   :error="hasError"
                   :valid="isValid"
                   :label="label"
                   :focus="isFocus"
                   :empty="isEmpty"
                   :required-marker="requiredMarker"
                   :icon-name="iconName"
                   :tag-style="tagStyle">
        <input ref="input"
               class="m-textfield__input"
               :disabled="!active"
               :placeholder="placeholder"
               :readonly="readonly"
               @focus="onFocus"
               @blur="onBlur"
               @keyup="onKeyup"
               @keydown="onKeydown"
               @change="onChange"
               @paste="onPaste"
               v-model="model">
        <div slot="right-content"
             class="m-textfield__icon-password"
             :class="{ 'm--is-transparent': !hasValue }"
             v-if="propPasswordIcon">
            <m-icon-button :icon-name="iconNamePassword"
                           button-size="22px"
                           :title="iconDescriptionPassword"
                           @click="togglePasswordVisibility"></m-icon-button>
        </div>
    </m-input-style>
    <m-validation-message
        :disabled="isDisabled"
        :waiting="isWaiting"
        :error="hasError"
        :valid="isValid"
        :error-message="errorMessage"
        :valid-message="validMessage"
        :helper-message="helperMessage"></m-validation-message>
</div>
