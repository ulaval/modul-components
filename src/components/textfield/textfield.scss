@import 'abstracts/variables';

.m-textfield {
    position: relative;
    display: inline-flex;
    flex-direction: column;

    &.m--has-error {
        .m-textfield__validation {
            color: $m-color--error;
        }
    }

    &.m--is-valid {
        .m-textfield__validation {
            color: $m-color--success;
        }
    }

    &__icon-password {
        margin-left: $m-margin--s;

        &.m--is-enter-active,
        &.m--is-leave-active {
            transition: opacity $m-transition-duration--s linear;
        }

        &.m--is-enter,
        &.m--is-leave-to {
            opacity: 0;
        }
    }

    &__validation {
        display: flex;
        justify-content: flex-end;

        &__message {
            flex: 1 1 auto;
        }

        &__maxlenght {
            transition: color $m-transition-duration ease;
            margin-top: 2px; // Magic number
            font-size: $m-font-size--xs;
            padding-left: $m-padding--xs;
        }
    }
}
