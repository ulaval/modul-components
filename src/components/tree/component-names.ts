export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
