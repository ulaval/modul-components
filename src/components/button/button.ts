import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './button.html?style=./button.scss';
import { BUTTON_NAME, ICON_NAME } from '../component-names';
import IconPlugin from '../icon/icon';
import SpinnerPlugin from '../spinner/spinner';

export enum MButtonType {
    Button = 'button',
    Submit = 'submit',
    Reset = 'reset'
}

export enum MButtonSkin {
    Primary = 'primary',
    Secondary = 'secondary'
}

export enum MButtonIconPosition {
    Left = 'left',
    Right = 'right'
}

@WithRender
@Component
export class MButton extends Vue {

    @Prop({
        default: MButtonType.Button,
        validator: value =>
            value == MButtonType.Button ||
            value == MButtonType.Submit ||
            value == MButtonType.Reset
    })
    public type: MButtonType;
    @Prop({
        default: MButtonSkin.Primary,
        validator: value =>
            value == MButtonSkin.Primary ||
            value == MButtonSkin.Secondary
    })
    public skin: MButtonSkin;
    @Prop()
    public precision: string;
    @Prop()
    public disabled: boolean;
    @Prop()
    public waiting: boolean;
    @Prop()
    public fullSize: boolean;
    @Prop()
    public iconName: string;
    @Prop({
        default: MButtonIconPosition.Left,
        validator: value =>
            value == MButtonIconPosition.Left ||
            value == MButtonIconPosition.Right
    })
    public iconPosition: MButtonIconPosition;
    @Prop({ default: '12px' })
    public iconSize: string;

    private onClick(event: Event): void {
        this.$emit('click', event);
        this.$el.blur();
    }

    private onFocus(event: Event): void {
        this.$emit('focus');
    }

    private onBlur(event: Event): void {
        this.$emit('blur');
    }

    private get isSkinPrimary(): boolean {
        return this.skin == MButtonSkin.Primary;
    }

    private get isSkinSecondary(): boolean {
        return this.skin == MButtonSkin.Secondary;
    }

    private get isWaiting(): boolean {
        return !this.disabled ? this.waiting : false;
    }

    private get hasIcone(): boolean {
        return !!this.iconName;
    }

    private get hasIconLeft(): boolean {
        return this.iconPosition == MButtonIconPosition.Left && this.hasIcone && !this.waiting;
    }

    private get hasIconRight(): boolean {
        return this.iconPosition == MButtonIconPosition.Right && this.hasIcone && !this.waiting;
    }

    private get hasWaitingIconLeft(): boolean {
        return this.iconPosition == MButtonIconPosition.Left && this.waiting;
    }

    private get hasWaitingIconRight(): boolean {
        return this.iconPosition == MButtonIconPosition.Right && this.waiting;
    }

    private get hasPrecision(): boolean {
        return !!this.precision || !!this.$slots.precision;
    }
}

const ButtonPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(BUTTON_NAME, 'plugin.install');
        v.use(IconPlugin);
        v.use(SpinnerPlugin);
        v.component(BUTTON_NAME, MButton);
    }
};

export default ButtonPlugin;
