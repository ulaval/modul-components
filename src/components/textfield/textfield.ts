import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';

import { InputLabel } from '../../mixins/input-label/input-label';
import { InputManagement } from '../../mixins/input-management/input-management';
import { InputState } from '../../mixins/input-state/input-state';
import { InputWidth } from '../../mixins/input-width/input-width';
import { ModulVue } from '../../utils/vue/vue';
import ButtonPlugin from '../button/button';
import { TEXTFIELD_NAME } from '../component-names';
import InputStyle from '../input-style/input-style';
import ValidationMesagePlugin from '../validation-message/validation-message';
import WithRender from './textfield.html?style=./textfield.scss';

export enum MTextfieldType {
    Text = 'text',
    Password = 'password',
    EMail = 'email',
    Url = 'url',
    Telephone = 'tel'
}

const ICON_NAME_PASSWORD_VISIBLE: string = 'show-password';
const ICON_NAME_PASSWORD_HIDDEN: string = 'hidden-password';

@WithRender
@Component({
    mixins: [
        InputState,
        InputManagement,
        InputWidth,
        InputLabel
    ]
})
export class MTextfield extends ModulVue {

    @Prop({
        default: MTextfieldType.Text,
        validator: value =>
            value === MTextfieldType.EMail ||
            value === MTextfieldType.Password ||
            value === MTextfieldType.Telephone ||
            value === MTextfieldType.Text ||
            value === MTextfieldType.Url
    })
    public type: MTextfieldType;
    @Prop({ default: true })
    public passwordIcon: boolean;

    private passwordAsText: boolean = false;

    private iconDescriptionShowPassword: string = this.$i18n.translate('m-textfield:show-password');
    private iconDescriptionHidePassword: string = this.$i18n.translate('m-textfield:hide-password');

    protected created(): void {
        if (!this.$i18n) {
            throw new Error('<m-text-field> -> this.$i18n is undefined, you must install the i18n plugin.');
        }
    }

    protected mounted(): void {
        (this.$refs.input as HTMLElement).setAttribute('type', this.inputType);
    }

    @Watch('type')
    private typeChanged(type: MTextfieldType): void {
        this.$log.warn('<' + TEXTFIELD_NAME + '>: Change of property "type" is not supported');
        (this.$refs.input as HTMLElement).setAttribute('type', this.inputType);
    }

    private togglePasswordVisibility(event): void {
        this.passwordAsText = !this.passwordAsText;
    }

    private get inputType(): MTextfieldType {
        let type: MTextfieldType = MTextfieldType.Text;
        if (this.type === MTextfieldType.Password && this.passwordAsText) {
            type = MTextfieldType.Text;
        } else if (this.type === MTextfieldType.Password || this.type === MTextfieldType.EMail || this.type === MTextfieldType.Url ||
            this.type === MTextfieldType.Telephone) {
            type = this.type;
        }
        this.$nextTick(() => {
            let inputEl: HTMLElement = this.$refs.input as HTMLElement;
            if (inputEl) {
                inputEl.setAttribute('type', type);
            }
        });
        return type;
    }

    private get iconNamePassword(): string {
        return this.passwordAsText ? ICON_NAME_PASSWORD_HIDDEN : ICON_NAME_PASSWORD_VISIBLE;
    }

    private get iconDescriptionPassword(): string {
        return this.passwordAsText ? this.iconDescriptionHidePassword : this.iconDescriptionShowPassword;
    }

    private get propPasswordIcon(): boolean {
        return this.passwordIcon && this.type === MTextfieldType.Password && this.as<InputState>().active;
    }
}

const TextfieldPlugin: PluginObject<any> = {
    install(v, options): void {
        v.use(InputStyle);
        v.use(ValidationMesagePlugin);
        v.use(ButtonPlugin);
        v.component(TEXTFIELD_NAME, MTextfield);
    }
};

export default TextfieldPlugin;
