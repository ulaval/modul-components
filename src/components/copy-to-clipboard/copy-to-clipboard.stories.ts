import { withA11y } from '@storybook/addon-a11y';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { MButton } from '../button/button';
import { COPY_TO_CLIPBOARD_NAME } from '../component-names';
import { MTextfield } from '../textfield/textfield';
import { COPY_TO_CLIPBOARD_FEEDBACK_NAME } from './../component-names';
import CopyToClipboardPlugin from './copy-to-clipboard';

Vue.use(CopyToClipboardPlugin);
const componentName: string = COPY_TO_CLIPBOARD_NAME;
const feedbackComponentName: string = COPY_TO_CLIPBOARD_FEEDBACK_NAME;

storiesOf(`${componentsHierarchyRootSeparator}${COPY_TO_CLIPBOARD_NAME}`, module)
    .addDecorator(withA11y)
    .add('default', () => ({
        template: `<${componentName} :value="''" />`
    }))
    .add('default with value', () => ({
        template: `<${componentName} :value="'some value'" />`
    }))
    .add('user feedback', () => ({
        template: `
        <${feedbackComponentName}>
            <template slot-scope="{ showToast }">
                <${componentName} :value="'some value'" @copy="showToast" />
            </template>
        </${feedbackComponentName}>`
    }))
    .add('custom user feedback', () => ({
        template: `
        <${feedbackComponentName} :message="'some message'" :position="'top-center'" :timeout="'none'">
            <template slot-scope="{ showToast }">
                <${componentName} :value="'some value'" @copy="showToast" />
            </template>
        </${feedbackComponentName}>`
    }))
    .add('custom', () => ({
        template: `
        <div>
            <${componentName} :value="value">
                <template slot-scope="{ inputProps, buttonHandlers }">
                    <m-textfield v-bind="inputProps" :label="'some label'">
                    </m-textfield>
                    <button v-on="buttonHandlers">Some button</button>
                </template>
            </${componentName}>
            <${componentName} :value="value">
                <template slot="inputButton" slot-scope="{ buttonHandlers }">
                    <button v-on="buttonHandlers">Some button</button>
                </template>
            </${componentName}>
        </div>`,
        data: () => ({
            value: 'some value'
        }),
        components: {
            MTextfield,
            MButton
        }
    }));
