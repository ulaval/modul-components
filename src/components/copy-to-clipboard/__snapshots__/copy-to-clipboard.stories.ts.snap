// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-copy-to-clipboard States 1`] = `
<div>
  <div>
    <div
      class="m-copy-to-clipboard"
    >
      <div
        class="m-textfield m--is-readonly m--is-type-text"
        style="width: 100%;"
      >
        <div
          class="m-input-style m--is-readonly m--is-tag-default"
          style="margin-top: 10px;"
        >
          <div
            class="m-input-style__main"
          >
            <div
              class="m-input-style__body"
            >
              <span
                class="m-input-style__label"
                style="z-index: -1;"
              >
                <span
                  class="m-input-style__text"
                />
              </span>
               
              <div
                class="m-input-style__input"
              >
                <div
                  class="m-input-style__prefix"
                />
                 
                <div
                  class="m-input-style__content"
                >
                  <input
                    class="m-textfield__input"
                    id="mTextfield-uuid"
                    maxlength="Infinity"
                    placeholder="Disabled"
                    readonly="readonly"
                    type="text"
                  />
                    
                  <!---->
                   
                  <!---->
                   
                  <!---->
                </div>
                 
                <div
                  class="m-input-style__suffix"
                >
                  <!---->
                   
                  <a
                    class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                    href="#"
                    role="button"
                    tabindex="0"
                  >
                    <!---->
                     
                    <span
                      class="m-link__text"
                    >
                      Copier
                    </span>
                     
                    <!---->
                  </a>
                   
                  <!---->
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="m-textfield__validation"
        >
          <div
            class="m-validation-message m-textfield__validation__message"
          >
            <!---->
             
            <!---->
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-copy-to-clipboard"
    >
      <div
        class="m-textfield m--is-readonly m--is-type-text"
        style="width: 100%;"
      >
        <div
          class="m-input-style m--is-readonly m--is-tag-default"
          style="margin-top: 10px;"
        >
          <div
            class="m-input-style__main"
          >
            <div
              class="m-input-style__body"
            >
              <span
                class="m-input-style__label"
                style="z-index: -1;"
              >
                <span
                  class="m-input-style__text"
                />
              </span>
               
              <div
                class="m-input-style__input"
              >
                <div
                  class="m-input-style__prefix"
                />
                 
                <div
                  class="m-input-style__content"
                >
                  <input
                    class="m-textfield__input"
                    id="mTextfield-uuid"
                    maxlength="Infinity"
                    placeholder="Read-only"
                    readonly="readonly"
                    type="text"
                  />
                    
                  <!---->
                   
                  <!---->
                   
                  <!---->
                </div>
                 
                <div
                  class="m-input-style__suffix"
                >
                  <!---->
                   
                  <a
                    class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                    href="#"
                    role="button"
                    tabindex="0"
                  >
                    <!---->
                     
                    <span
                      class="m-link__text"
                    >
                      Copier
                    </span>
                     
                    <!---->
                  </a>
                   
                  <!---->
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="m-textfield__validation"
        >
          <div
            class="m-validation-message m-textfield__validation__message"
          >
            <!---->
             
            <!---->
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-copy-to-clipboard"
    >
      <div
        class="m-textfield m--is-waiting m--is-type-text"
        style="width: 100%;"
      >
        <div
          class="m-input-style m--is-waiting m--is-tag-default"
          style="margin-top: 10px;"
        >
          <div
            class="m-input-style__main"
          >
            <div
              class="m-input-style__body"
            >
              <span
                class="m-input-style__label"
                style="z-index: -1;"
              >
                <span
                  class="m-input-style__text"
                />
              </span>
               
              <div
                class="m-input-style__input"
              >
                <div
                  class="m-input-style__prefix"
                />
                 
                <div
                  class="m-input-style__content"
                >
                  <input
                    class="m-textfield__input"
                    disabled="disabled"
                    id="mTextfield-uuid"
                    maxlength="Infinity"
                    placeholder="Waiting"
                    type="text"
                  />
                    
                  <!---->
                   
                  <!---->
                   
                  <!---->
                </div>
                 
                <div
                  class="m-input-style__suffix"
                >
                  <!---->
                    
                  <div
                    class="m-spinner m-input-style__spinner"
                  >
                    <!---->
                     
                    <div />
                  </div>
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="m-textfield__validation"
        >
          <div
            class="m-validation-message m-textfield__validation__message"
          >
            <!---->
             
            <!---->
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <button>
    Test
  </button>
</div>
`;

exports[`Storyshots components|m-copy-to-clipboard custom 1`] = `
<div>
  <div
    class="m-copy-to-clipboard"
  >
    <div
      class="m-textfield m--is-readonly m--has-label m--is-type-text"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-readonly m--is-label-up m--has-label m--has-value m--is-tag-default"
        style="margin-top: 10px;"
      >
        <div
          class="m-input-style__main"
        >
          <div
            class="m-input-style__body"
          >
            <label
              class="m-input-style__label"
              for="mTextfield-uuid"
              style="margin-right: 0px;"
            >
              <span
                class="m-input-style__text"
              >
                <span>
                  some label
                </span>
                Â 
                <!---->
              </span>
            </label>
             
            <div
              class="m-input-style__input"
            >
              <!---->
               
              <div
                class="m-input-style__content"
              >
                <input
                  class="m-textfield__input"
                  id="mTextfield-uuid"
                  maxlength="Infinity"
                  readonly="readonly"
                  type="text"
                />
                  
                <!---->
                 
                <!---->
                 
                <!---->
              </div>
               
              <div
                class="m-input-style__suffix"
              >
                <!---->
                  
                <!---->
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <div
        class="m-textfield__validation"
      >
        <div
          class="m-validation-message m-textfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
     
    <button>
      Some button
    </button>
  </div>
   
  <div
    class="m-copy-to-clipboard"
  >
    <div
      class="m-textfield m--is-readonly m--is-type-text"
      style="width: 100%;"
    >
      <div
        class="m-input-style m--is-readonly m--has-value m--is-tag-default"
        style="margin-top: 10px;"
      >
        <div
          class="m-input-style__main"
        >
          <div
            class="m-input-style__body"
          >
            <span
              class="m-input-style__label"
              style="z-index: -1;"
            >
              <span
                class="m-input-style__text"
              />
            </span>
             
            <div
              class="m-input-style__input"
            >
              <div
                class="m-input-style__prefix"
              />
               
              <div
                class="m-input-style__content"
              >
                <input
                  class="m-textfield__input"
                  id="mTextfield-uuid"
                  maxlength="Infinity"
                  readonly="readonly"
                  type="text"
                />
                  
                <!---->
                 
                <!---->
                 
                <!---->
              </div>
               
              <div
                class="m-input-style__suffix"
              >
                <!---->
                 
                <button>
                  Some button
                </button>
                 
                <!---->
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <div
        class="m-textfield__validation"
      >
        <div
          class="m-validation-message m-textfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-copy-to-clipboard custom user feedback 1`] = `
<div
  class="m-copy-to-clipboard"
>
  <div
    class="m-textfield m--is-readonly m--is-type-text"
    style="width: 100%;"
  >
    <div
      class="m-input-style m--is-readonly m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <input
                class="m-textfield__input"
                id="mTextfield-uuid"
                maxlength="Infinity"
                readonly="readonly"
                type="text"
              />
                
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
              <!---->
               
              <a
                class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                href="#"
                role="button"
                tabindex="0"
              >
                <!---->
                 
                <span
                  class="m-link__text"
                >
                  Copier
                </span>
                 
                <!---->
              </a>
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textfield__validation"
    >
      <div
        class="m-validation-message m-textfield__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-copy-to-clipboard default 1`] = `
<div
  class="m-copy-to-clipboard"
>
  <div
    class="m-textfield m--is-readonly m--is-type-text"
    style="width: 100%;"
  >
    <div
      class="m-input-style m--is-readonly m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <input
                class="m-textfield__input"
                id="mTextfield-uuid"
                maxlength="Infinity"
                readonly="readonly"
                type="text"
              />
                
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
              <!---->
               
              <a
                class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                href="#"
                role="button"
                tabindex="0"
              >
                <!---->
                 
                <span
                  class="m-link__text"
                >
                  Copier
                </span>
                 
                <!---->
              </a>
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textfield__validation"
    >
      <div
        class="m-validation-message m-textfield__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-copy-to-clipboard default with value 1`] = `
<div
  class="m-copy-to-clipboard"
>
  <div
    class="m-textfield m--is-readonly m--is-type-text"
    style="width: 100%;"
  >
    <div
      class="m-input-style m--is-readonly m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <input
                class="m-textfield__input"
                id="mTextfield-uuid"
                maxlength="Infinity"
                readonly="readonly"
                type="text"
              />
                
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
              <!---->
               
              <a
                class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                href="#"
                role="button"
                tabindex="0"
              >
                <!---->
                 
                <span
                  class="m-link__text"
                >
                  Copier
                </span>
                 
                <!---->
              </a>
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textfield__validation"
    >
      <div
        class="m-validation-message m-textfield__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-copy-to-clipboard user feedback 1`] = `
<div
  class="m-copy-to-clipboard"
>
  <div
    class="m-textfield m--is-readonly m--is-type-text"
    style="width: 100%;"
  >
    <div
      class="m-input-style m--is-readonly m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <input
                class="m-textfield__input"
                id="mTextfield-uuid"
                maxlength="Infinity"
                readonly="readonly"
                type="text"
              />
                
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
              <!---->
               
              <a
                class="m-link m-copy-to-clipboard__button m--is-unvisited m--no-underline m--is-button m--has-content"
                href="#"
                role="button"
                tabindex="0"
              >
                <!---->
                 
                <span
                  class="m-link__text"
                >
                  Copier
                </span>
                 
                <!---->
              </a>
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textfield__validation"
    >
      <div
        class="m-validation-message m-textfield__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;
